Sources/Kitura/Kitura.swift

```diff
@@ -113,10 +114,24 @@ public class Kitura {
     /// Kitura.addHTTPServer(onPort: 8080, with: router)
     /// Kitura.run()
     ///```
+    /// Make all registered servers start listening on their port and exit if any fail to start.
+    ///```swift
+    /// let router = Router()
+    /// Kitura.addHTTPServer(onPort: 8080, with: router)
+    /// Kitura.run(withStatus: true)
+    ///```
     /// - note: This function never returns - it should be the last call in your `main.swift` file.
-    public class func run() {
+    /// - Parameter withStatus: Determines whether the Kitura process can return a non-zero exit code should any of the servers fail to start. Defaults to false, indicating it will not return any non-zero exit codes.
+    public class func run(withStatus status: Bool = false) {
```

Comment:
I don't like the name `withStatus` here, as this function doesn't return an exit code, it calls `exit` if a failure occurs.  How about `exitOnFailure`?

I also think there should be a `run()` variant that does return the status code, eg. `runWithStatus() -> Int`, so that a user has the option of handling a situation where one or more servers failed to start.  This function should have a parameter that controls whether run waits if at least one listener is successful, or if it should return immediately if at least one failure occurs.

Reply:
I'm not sure about calling `exit()` at all... Can we return a `@discardableResult` from `run()` instead and let the user decide what to do?

Reply:
@ianpartridge if we add an `@discardableResult` to the existing `run()` then the function no longer has the same return type and would be a breaking change.  Now granted, this is _unlikely_ to break users, but it would if someone wrote this:
```swift
func startMyServer() -> Void {
  return Kitura.run()
}
```
Unfortunately you can't overload `run()` by return type only.  Well, you _can_, but that's also breaking because people who just write `Kitura.run()` will then get a compilation failure because the statement is ambiguous.

Reply:
I don't like the parameter name `withStatus` either but it's what was suggested in the issue so I used it as I couldn't decide on anything better. 

I don't really like `exitOnFailure ` either. What about:

`run(withExit: true)`
`run(asExitable: true)`
`run(inExitMode: true)`

Or something like that, basically something that reads `Swift-like`. 

Rather then adding another public function, could be not add a property that keeps track of how many failures occurred? 

Could look something like this:

```swift
run(withStatus = true, canFail = true) //Needs a better name
```
Then inside `startWithStatus` we can check whether those values have been set and respond appropriately. 

- If `true, true`:  return immediately if at least one failure occurs.
- if `true, false`: wait to see if at least one listener is successful.
- if `false, any`: Can't exit so ignore second param. 

Just an idea, I haven't tested it yet nor am I aware of any implications of implementing it this way.

Reply:
`exitOnFailure`?

Reply:
Oh Dave already suggested it! Sounds least-worst to me...